src/test/java/com/wikia/webdriver/testcases/mercurytests/HeaderTest.java

```diff
@@ -0,0 +1,147 @@
+
+  private static final String ELEMENT_EXPECTATION_MESSAGE_TEMPLATE = "%s was expected to be %s.";
+  private static final String INVALID_ATTRIBUTE_MESSAGE_TEMPLATE = "%s has invalid %s attribute.";
+
+
+  @Test(groups = "mercury_header_checkElementsVisibilityWithoutInfobox")
+  public void mercury_header_checkElementsVisibilityWithoutInfobox() {
+    Header header =
+        new ArticlePage()
+            .open(MercurySubpages.NO_INFOBOX)
+            .getHeader();
+
+    Assertion.assertTrue(
+        header.isHeaderVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, HEADER_MESSAGE, VISIBLE_MESSAGE)
+    );
+    Assertion.assertTrue(
+        header.isPageTitleVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, PAGE_TITLE_MESSAGE, VISIBLE_MESSAGE)
+    );
+    Assertion.assertFalse(
+        header.isHeroImageVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, HERO_IMAGE_MESSAGE, INVISIBLE_MESSAGE)
+    );
+  }
+
+  @Test(groups = "mercury_header_checkElementsVisibilityWithInfoboxAndWithHeroImage")
+  public void mercury_header_checkElementsVisibilityWithInfoboxAndWithHeroImage() {
+    Header header =
+        new ArticlePage()
+            .open(MercurySubpages.INFOBOX_1)
+            .getHeader();
+
+    Assertion.assertTrue(
+        header.isHeaderVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, HEADER_MESSAGE, VISIBLE_MESSAGE)
+    );
+    Assertion.assertTrue(
+        header.isPageTitleVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, PAGE_TITLE_MESSAGE, VISIBLE_MESSAGE)
+    );
+    Assertion.assertTrue(
+        header.isHeroImageVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, HERO_IMAGE_MESSAGE, VISIBLE_MESSAGE)
+    );
+  }
+
+  @Test(groups = "mercury_header_checkElementsVisibilityWithInfoboxAndWithoutHeroImage")
+  public void mercury_header_checkElementsVisibilityWithInfoboxAndWithoutHeroImage() {
+    Header header =
+        new ArticlePage()
+            .open(MercurySubpages.INFOBOX_2)
+            .getHeader();
+
+    Assertion.assertTrue(
+        header.isHeaderVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, HEADER_MESSAGE, VISIBLE_MESSAGE)
+    );
+    Assertion.assertTrue(
+        header.isPageTitleVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, PAGE_TITLE_MESSAGE, VISIBLE_MESSAGE)
+    );
+    Assertion.assertFalse(
+        header.isHeroImageVisible(),
+        String.format(ELEMENT_EXPECTATION_MESSAGE_TEMPLATE, HERO_IMAGE_MESSAGE, INVISIBLE_MESSAGE)
+    );
+  }
+
+  @Test(groups = "mercury_header_heroImageIsProperlyStyled")
+  public void mercury_header_heroImageIsProperlyStyled() {
+    Header header =
+        new ArticlePage()
+            .open(MercurySubpages.INFOBOX_1)
+            .getHeader();
+
+    Map<String, String> attributeExpectedValues = new HashMap<String, String>() {{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1171)